src/consensus.rs

```diff
@@ -313,14 +312,20 @@ impl <L, M> Consensus<L, M> where L: Log, M: StateMachine {
                         if existing_term != leader_prev_log_term {
                             scoped_debug!("AppendEntriesRequest: inconsistent previous log term: leader: {}, local: {}",
                                           leader_prev_log_term, existing_term);
-                            messages::append_entries_response_inconsistent_prev_entry(self.current_term())
+                            // If an existing entry conflicts with a new one (same index but different terms),
+                            // delete the existing entry and all that follow it
+                            messages::append_entries_response_inconsistent_prev_entry(self.current_term(),
+                                leader_prev_log_index)
```

Comment:
answered by @danburkert:

>
[9:05pm] tschottdorf: dcb: let from_index = Into::<u64>::into(self.leader_state.next_index(&peer));
[9:06pm] tschottdorf: I don’t see anything that initializes next_index(&peer) except as a reaction to an append. What is it on the first append?
[9:06pm] dcb: tschottdorf: as per the paper, it's set to the leader's latest index + 1 when the leader is elected
[9:06pm] dcb: this happens in state.rs
[9:09pm] tschottdorf: ah, that’s what I was looking for.
[9:09pm] tschottdorf: ok, I’m with you now

So all is well.